apps/fishing-map/features/map/popups/VesselsTable.tsx

```diff
@@ -99,70 +134,111 @@ function VesselsTable({
+          {vessels?.length > 0 &&
+            vessels.map((vessel, i) => {
+              const vesselName =
+                vessel.shipname !== undefined
+                  ? formatInfoField(vessel.shipname, 'name')
+                  : t('vessel.notMatched', 'No data matched')
+
+              const vesselGearType = `${t(
+                `vessel.gearTypes.${vessel.geartype}` as any,
+                EMPTY_FIELD_PLACEHOLDER
+              )}`
+
+              const hasDatasets =
+                vessel.infoDataset !== undefined || vessel.trackDataset !== undefined
+
+              const vesselInWorkspace = getVesselInWorkspace(vessels, vessel.id)
+
+              const pinTrackDisabled = !interactionAllowed || !hasDatasets
+              return (
+                <tr key={i}>
+                  {!pinTrackDisabled && (
+                    <td className={styles.icon}>
+                      <IconButton
+                        icon={vesselInWorkspace ? 'pin-filled' : 'pin'}
+                        style={{
+                          color: vesselInWorkspace ? vesselInWorkspace.config.color : '',
+                        }}
+                        tooltip={
+                          vesselInWorkspace
+                            ? t(
+                                'search.vesselAlreadyInWorkspace',
+                                'This vessel is already in your workspace'
+                              )
+                            : t('search.seeVessel', 'See vessel')
+                        }
+                        onClick={(e) => onVesselClick(e, vessel)}
+                        size="small"
+                      />
+                    </td>
+                  )}
+                  <td colSpan={hasPinColumn && pinTrackDisabled ? 2 : 1}>{vesselName}</td>
+                  <td className={styles.columnSpace}>
+                    <Tooltip content={t(`flags:${vessel.flag as string}` as any)}>
+                      <span>{vessel.flag || EMPTY_FIELD_PLACEHOLDER}</span>
+                    </Tooltip>
+                  </td>
+                  <td className={styles.columnSpace}>{vesselGearType}</td>
+                  <td className={styles.columnSpace}>{vessel.dataset && vessel.dataset.name}</td>
+                  <td className={cx(styles.vesselsTableHour, styles.columnSpace)}>
+                    <I18nNumber number={vessel[vesselProperty]} />
+                  </td>
+                </tr>
+              )
+            })}
+        </tbody>
+      </table>
+      {feature.vesselsInfo && !showFullList && (
+        <Fragment>
+          {feature.vesselsInfo.overflow && (
+            <button
+              className={styles.vesselsMore}
+              disabled={!gfwUser}
+              onClick={() => setModalOpen(true)}
+            >
+              + {feature.vesselsInfo.overflowNumber} {t('common.more', 'more')}
+            </button>
+          )}
+        </Fragment>
+      )}
+      {gfwUser && (
+        <Modal
```

Comment:
I'm really not sure about moving the modal logic inside `VesselsTable`. Correct me if I'm wrong, but it means when you instanciate VesselsTable, it instanciates itself within the modal? Which would mean that the modal instanciates another modal for no reason. Is it the reason why the vessels count label at the bottom of the table inside the modal wrongly has a pointer cursor? Maybe worth discussing as I'm not sure what's happening here but it smells like a smell.

Reply:
Could be structured this way:
```
FishingRow / Viirs Row
  -> VesselsTablePopupRow
    -> VesselsTable
    -> Modal
      -> VesselsTable  

```

Reply:
Good catch with the double Modal instances so I've fixed it with this [commit](https://github.com/GlobalFishingWatch/frontend/commit/921775e64c564ce40d6b986a4eca7dbd5fef2406)

Because I think is it much simpler to handle the modal within the own table component that replicating in every PopupRow used